### PR TITLE
Autocomplete loadout hashtags in loadout drawer

### PR DESCRIPTION
**Pull request by robojumper**
Pulled out of #8916. The textcomplete package doesn't support `<input />` out of the box, but the code basically works fine even when the types don't say it, so a cast makes it work.

grafik
grafik

### CODE REVIEW

**Comment by robojumper** [comment]
So `textcomplete`'s `TextareaEditor` is written for a `<textarea />` but it seems (Edge, Firefox) to also work fine with an `<input type="text" />`, so lying about the types works. The code is spread across three different packages in three different repos, so upstreaming `<input />` support is not that simple. Not sure what the best approach is -- we could also just copy a bunch of code into DIM but it'd basically need no changes other than type signature adjustments.

**Comment by bhollis** [comment]
Hmm - we used to use this for autocomplete in the main search bar, which was a regular text input. But maybe that was a previous version of the library.

File: src/app/dim-ui/text-complete/text-complete.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ */
+export function useAutocomplete(
+  textArea: React.RefObject<HTMLTextAreaElement | HTMLInputElement>,
+  tags: string[]
+) {
+  useEffect(() => {
+    if (textArea.current) {
+      // commit a type crime here because textcomplete says it only works with
+      // TextArea but happens to also work entirely fine with Input[type=text]
+      const editor = new TextareaEditor(textArea.current as unknown as HTMLTextAreaElement);
```
Comment (bhollis) [inline src/app/dim-ui/text-complete/text-complete.ts:56]: Sub to https://github.com/yuku/textcomplete/issues/355

**Review by bhollis — APPROVED** [review]
Very cool

**Pull request merged**